lib/csv/table.rb

```diff
@@ -365,14 +365,91 @@ def headers
       end
     end
 
+    # :call-seq:
+    #   table[n] -> row
+    #   table[range] -> array_of_rows
+    #   table[header] -> array_of_fields
     #
-    # In the default mixed mode, this method returns rows for index access and
-    # columns for header access. You can force the index association by first
-    # calling by_col!() or by_row!().
+    # Returns data from the table;  does not modify the table.
+    #
+    # ---
+    #
+    # The expression <tt>table[n]</tt>, where +n+ is a non-negative \Integer,
+    # returns the +n+th row of the table, if that row exists,
+    # and if the access mode is <tt>:row</tt> or <tt>:col_or_row</tt>:
+    #   source = "Name,Value\nfoo,0\nbar,1\nbaz,2\n"
+    #   table = CSV.parse(source, headers: true)
+    #   table.by_row! # => #<CSV::Table mode:row row_count:4>
+    #   table[1] # => #<CSV::Row "Name":"bar" "Value":"1">
+    #   table.by_col_or_row! # => #<CSV::Table mode:col_or_row row_count:4>
+    #   table[1] # => #<CSV::Row "Name":"bar" "Value":"1">
+    #
+    # Counts backward from the last row if +n+ is negative:
+    #   table[-1] # => #<CSV::Row "Name":"baz" "Value":"2">
+    #
+    # Returns +nil+ if +n+ is too large or too small:
+    #   table[4] # => nil
+    #   table[-4] => nil
+    #
+    # Raises an exception if the access mode is <tt>:row</tt>
+    # and +n+ is not an
+    # {Integer-convertible object}[doc/implicit_conversion_rdoc.html#label-Integer-Convertible+Objects].
```

Comment:
We can't refer it by `doc/...` because we may generate document only for csv.
How about using https://docs.ruby-lang.org/en/master/implicit_conversion_rdoc.html ?

Reply:
Oh, excellent!  Done!